### DISCUSSION TITLE
Toast: How to find out when a user has closed Toast

### DISCUSSION THREAD

**Discussion by DKPlugins** [primefaces/community#3177] [PrimeReact] [answered]
How can I find out when the user closed the Toast, and when it automatically disappeared after the time expired?

**Comment by LeandroMarcondes** [comment, marked as answer]
You can use the callback onRemove to monitor when Toad is hide in general. Then use the pt > closeButton, to add an onClick event so you can monitor when the user actively clicked on the button. Here is a sample:

[Sample](https://stackblitz.com/edit/m57irumd?file=src%2FApp.jsx)

```
<Toast
      ref={toast}
      onHide={onHide}
      onRemove={onRemove}
      pt={{
        closeButton: {
          onClick: () => {
            onClickClose();
          },
        },
      }}
/>
```
